src/autopas/LogicHandler.h

```diff
@@ -1772,42 +1771,93 @@ void LogicHandler<Particle_T>::remainderHelper3bBufferBufferContainerAoS(
 template <class Particle_T>
 template <bool newton3, class ContainerType, class TriwiseFunctor>
 void LogicHandler<Particle_T>::remainderHelper3bBufferContainerContainerAoS(
-    const std::vector<Particle_T *> &bufferParticles, const size_t numOwnedBufferParticles, ContainerType &container,
-    TriwiseFunctor *f) {
-  // Todo: parallelize without race conditions - https://github.com/AutoPas/AutoPas/issues/904
-  using autopas::utils::ArrayUtils::static_cast_copy_array;
+    const std::vector<Particle_T *> &bufferParticles, ContainerType &container, TriwiseFunctor *f) {
   using namespace autopas::utils::ArrayMath::literals;
-
+  if (bufferParticles.empty()) {
+    return;
+  }
+  const size_t numBufferParticles = bufferParticles.size();
   const double cutoff = container.getCutoff();
+  const double cutoffSquared = cutoff * cutoff;
+
+  // Step 1: Create a sorted cell view on all buffer particles
+  const auto domainDiagonal = container.getBoxMax() - container.getBoxMin();
+  ReferenceParticleCell<Particle_T> cellView;
+  cellView.reserve(numBufferParticles);
+  for (auto pRef : bufferParticles) {
+    cellView.addParticleReference(pRef);
+  }
+  SortedCellView<ReferenceParticleCell<Particle_T>> sortedCellView(cellView, domainDiagonal);
 
-  for (auto i = 0; i < bufferParticles.size(); ++i) {
-    Particle_T &p1 = *bufferParticles[i];
-    const auto pos = p1.getR();
-    const auto boxmin = pos - cutoff;
-    const auto boxmax = pos + cutoff;
+  // Step 2: Create a neighbor list of pairs for each of the buffer particles.
+  // This step accesses all particles in read-only mode and can be fully parallelized.
+  std::vector<std::vector<std::pair<Particle_T *, Particle_T *>>> bufferParticleNeighborLists(numBufferParticles);
+
+  AUTOPAS_OPENMP(parallel for schedule(static))
```

Comment:
Fyi, I suspect something other than `static` would be more suitable as there is probably a fair bit of load imbalance. But I also think this is fine to merge as is (better than nothing).